src/cpp/fastdds/xtypes/utils.cpp

```diff
@@ -0,0 +1,1031 @@
+                if (RETCODE_OK == ret)
+                {
+                    json_insert(member_name, j_map, output);
+                }
+                // Return loaned value
+                ReturnCode_t ret_return_loan;
+                if (RETCODE_OK != (ret_return_loan = data->return_loaned_value(st_data)))
+                {
+                    EPROSIMA_LOG_WARNING(XTYPES_UTILS, "Error encountered while returning loaned value.");
+                }
+                // Give priority to prior error if occurred
+                if (RETCODE_OK != ret)
+                {
+                    return ret;
+                }
+                else
+                {
+                    return ret_return_loan;
+                }
+            }
+            else
+            {
+                EPROSIMA_LOG_WARNING(XTYPES_UTILS,
+                        "Error encountered while serializing map member to JSON: loan_value failed.");
+                return RETCODE_BAD_PARAMETER;
+            }
+        }
+        case TK_ALIAS:
+        {
+            EPROSIMA_LOG_WARNING(XTYPES_UTILS,
+                    "Error encountered while serializing member to JSON: unexpected TK_ALIAS kind.");
+            return RETCODE_BAD_PARAMETER;
+        }
+        default:
+            EPROSIMA_LOG_WARNING(XTYPES_UTILS,
+                    "Error encountered while serializing map member to JSON: unexpected kind " << member_kind <<
+                    " found.");
+            return RETCODE_BAD_PARAMETER;
+    }
+}
+
+ReturnCode_t json_serialize_member(
+        const traits<DynamicDataImpl>::ref_type& data,
+        MemberId member_id,
+        TypeKind member_kind,
+        nlohmann::json& output,
+        DynamicDataJsonFormat format) noexcept
+{
+    return json_serialize_member(data, member_id, member_kind, "", output, format);
+}
+
+ReturnCode_t json_serialize_basic_member(
+        const traits<DynamicDataImpl>::ref_type& data,
+        MemberId member_id,
+        TypeKind member_kind,
+        const std::string& member_name,
+        nlohmann::json& output,
+        DynamicDataJsonFormat format) noexcept
+{
+    switch (member_kind)
+    {
+        case TK_NONE:
+        {
+            EPROSIMA_LOG_WARNING(XTYPES_UTILS,
+                    "Error encountered while serializing basic member to JSON: unexpected TK_NONE kind.");
+            return RETCODE_BAD_PARAMETER;
+        }
+        case TK_BOOLEAN:
+        {
+            ReturnCode_t ret;
+            bool value;
+            if (RETCODE_OK == (ret = data->get_boolean_value(value, member_id)))
+            {
+                json_insert(member_name, value, output);
+            }
+            else
+            {
+                EPROSIMA_LOG_WARNING(XTYPES_UTILS, "Error encountered while serializing TK_BOOLEAN member to JSON.");
+            }
+            return ret;
+        }
+        case TK_BYTE:
+        {
+            ReturnCode_t ret;
+            eprosima::fastrtps::rtps::octet value;
```

Comment:
Please, remove the namespace `eprosima`

Reply:
Although I believe they will soon be moving this to fastdds::dds.